src/main/scala/viper/gobra/frontend/PackageResolver.scala

```diff
@@ -230,22 +230,51 @@ object PackageResolver {
     } yield pkgName
   }
 
-  private lazy val pkgClauseRegex = """(?:\/\/.*|\/\*(?:.|\n)*\*\/|package(?:\s|\n)+([a-zA-Z_][a-zA-Z0-9_]*))""".r
+  private lazy val pkgClauseRegex = """(?:\/\/.*|\/\*(?:.|\n)*?\*\/|package(?:\s|\n)+([a-zA-Z_][a-zA-Z0-9_]*))""".r
 
-  private def getPackageClause(file: InputResource): Option[String] = {
-    val inputStream = file.asStream()
-    val bufferedSource = new BufferedSource(inputStream)
-    val content = bufferedSource.mkString
-    bufferedSource.close()
+  def getPackageClause(src: Source): Option[String] = {
 
     // TODO is there a way to perform the regex lazily on the file's content?
     pkgClauseRegex
-      .findAllMatchIn(content)
+      .findAllMatchIn(src.content)
       .collectFirst { case m if m.group(1) != null => m.group(1) }
   }
 
+  def getPackageId(packageName: String, src: Source, includeDirs: Vector[Path]): String = {
+    /**
+     * Relativizes the given path with the specified include directories or the go path. The resulting path is relative
+     * to the first include directory, that is a parent of it.
+     */
+    def relativizePath(path: Path): Path = {
+      val allDirs = Properties.envOrNone("GOPATH").map(path => Path.of(path)) match {
```

Comment:
What do you do if `GOPATH` is a semicolon separated list of paths?

Reply:
Good catch, I didn't have in mind that it probably isn't just a single path